python/dgl/transform.py

```diff
@@ -403,4 +404,84 @@ def laplacian_lambda_max(g):
                                       return_eigenvectors=False)[0].real)
     return rst
 
+
+def onehot_degree(g, max_degree=-1, out_field='d', direction="in"):
+    """Inplace add one-hot degree vector as node feature
+
+    Parameters
+    -----------
+    g: DGLGraph
+    max_degress: int
+        Maximum degree for one-hot encoding. If it's -1,
+        the maximum degree would be infered from the input graph.
+    out_field: str
+        Field name for the node feature
+    direction: str
+        Either "in" or "out". Specify whether to use in degrees or out degrees.
+    """
+    if direction == "in":
+        degrees = g.in_degrees()
+    elif direction == "out":
+        degrees = g.out_degrees()
+    else:
+        raise RuntimeError("Invalid Direction")
+    g.ndata[out_field] = F.one_hot(degrees, max_degree)
+
+def to_self_loop(g):
```

Comment:
The operation seems to be in-place. Previously all transforms related to graph topology are not in-place.